server/kafka/src/main/java/org/hawkular/apm/server/kafka/AbstractConsumerKafka.java

```diff
@@ -0,0 +1,224 @@
+     *
+     * @param topic The topic
+     * @param groupId The consumer group id
+     */
+    protected void init(String topic, String groupId) {
+        Properties props = new Properties();
+        props.put("bootstrap.servers", PropertyUtil.getProperty(PropertyUtil.HAWKULAR_APM_KAFKA_SERVERS,
+                "localhost:9092"));
+        props.put("group.id", groupId);
+        props.put("enable.auto.commit", "true");
+        props.put("auto.commit.interval.ms",
+                PropertyUtil.getProperty(PropertyUtil.HAWKULAR_APM_KAFKA_CONSUMER_AUTO_COMMIT_INTERVAL, "1000"));
+        props.put("session.timeout.ms",
+                PropertyUtil.getProperty(PropertyUtil.HAWKULAR_APM_KAFKA_CONSUMER_SESSION_TIMEOUT, "30000"));
+        props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
+
+        String maxPollRecords = PropertyUtil.getProperty(PropertyUtil.HAWKULAR_APM_KAFKA_MAX_POLL_RECORDS);
+        if (maxPollRecords != null) {
+            props.put("max.poll.records", maxPollRecords);
+        }
+
+        consumer = new KafkaConsumer<>(props);
+        consumer.subscribe(Arrays.asList(topic));
+
+        pollingInterval = PropertyUtil.getPropertyAsInteger(PropertyUtil.HAWKULAR_APM_KAFKA_POLLING_INTERVAL,
+                DEFAULT_POLLING_INTERVAL);
+    }
+
+    /**
+     * @return the typeReference
+     */
+    public TypeReference<S> getTypeReference() {
+        return typeReference;
+    }
+
+    /**
+     * @param typeReference the typeReference to set
+     */
+    public void setTypeReference(TypeReference<S> typeReference) {
+        this.typeReference = typeReference;
+    }
+
+    /**
+     * @return the processor
+     */
+    public Processor<S, T> getProcessor() {
+        return processor;
+    }
+
+    /**
+     * @param processor the processor to set
+     */
+    public void setProcessor(Processor<S, T> processor) {
+        this.processor = processor;
+    }
+
+    /**
+     * @return the publisher
+     */
+    public Publisher<T> getPublisher() {
+        return publisher;
+    }
+
+    /**
+     * @param publisher the publisher to set
+     */
+    public void setPublisher(Publisher<T> publisher) {
+        this.publisher = publisher;
+    }
+
+    /**
+     * @return the retryPublisher
+     */
+    public Publisher<S> getRetryPublisher() {
+        return retryPublisher;
+    }
+
+    /**
+     * @param retryPublisher the retryPublisher to set
+     */
+    public void setRetryPublisher(Publisher<S> retryPublisher) {
+        this.retryPublisher = retryPublisher;
+    }
+
+    /**
+     * This method returns the polling interval for the processor.
+     *
+     * @return The polling interval
+     */
+    protected long getPollingInterval() {
+        return pollingInterval;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
```

Comment:
This class is executed on standalone or in a server ?
Wondering if it could good to have some exit condition of the loop, for example, some flag that can be set on a @PostDestroy method to allow the thread stop friendly on a server shutdown.